### ISSUE TITLE
protobuf plugin

### ISSUE THREAD

**Issue by cmkarlsson** [basho/rebar3_protobuf_plugin#1]
I wasn't aware of this repository before. I had to create a rebar3 protobuf plugin to be able to compile the riak client (which has riak_pb as dependency).  You can find it under https://github.com/cmkarlsson/rebar3_protobuffs_plugin if there is something you want to reuse here.

Cheers

**Comment by binarytemple-external** [comment]
@lukebakken @cuyler I've tested @cmkarlsson plugin against riak-erlang-client (master 13 Dec 2016) and it builds. This project seems abandoned. Perhaps it would be good to adopt @cmkarlsson project as offered?

**Comment by lukebakken** [comment]
I'm looking into why this repository exists.

**Comment by lukebakken** [comment]
For what it's worth, I have switched the erlang client to use `gpb` instead of `erlang_protobuffs` and modified the build process so that a protobuf compiler is not run on every build, making these plugins a moot point "very soon now" 😄 

See basho/riak-erlang-client#339

**Comment by cuyler** [comment]
@lukebakken It was an abandoned attempt at rebar3 conversion two summers ago.

**Comment by binarytemple-external** [comment]
@lukebakken not sure what the implications of gpb will be, will it build with rebar3 ?

**Comment by lukebakken** [comment]
Yes, because the `.proto` files are parsed *only* when changes to them are made, and the generated files are commited to the repo ([`erl_protogen` target](https://github.com/basho/riak_pb/blob/develop/Makefile#L27-L30)). Prior to my change, the `.proto` files are parsed on every build which requires a `rebar` plugin.

**Comment by binarytemple-external** [comment]
@lukebakken  I mean, will it still be possible to build riak-erlang-client using rebar3 with @cmkarlsson plugin. Or will the changes break that?

**Comment by lukebakken** [comment]
As of commit adc603b in `riak_pb` and c203271b in `riak-erlang-client`, you do *not* need *any* protobuf plugins for `rebar` to build either project. Go ahead and give it a try - the `develop` branch in both projects should work.

**Comment by binarytemple-external** [comment]
@lukebakken thanks, will give it a try.
